components/search/src/types.rs

```diff
@@ -0,0 +1,197 @@
+    }
+}
+
+/// The list of possible update channels for a user's build.
+/// Use `default` for a self-build or an unknown channel.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchUpdateChannel {
+    Default,
+    Nightly,
+    Aurora,
+    Beta,
+    Release,
+    ESR,
+}
+
+impl SearchUpdateChannel {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            SearchUpdateChannel::Default => "default",
+            SearchUpdateChannel::Nightly => "nightly",
+            SearchUpdateChannel::Aurora => "aurora",
+            SearchUpdateChannel::Beta => "beta",
+            SearchUpdateChannel::Release => "release",
+            SearchUpdateChannel::ESR => "esr",
+        }
+    }
+}
+
+/// The user's environment that is used for filtering the search configuration.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchUserEnvironment {
+    /// The current locale of the application that the user is using.
+    pub locale: String,
+
+    /// The home region that the user is currently identified as being within.
+    /// On desktop & android there is a 14 day lag after detecting a region
+    /// change before the home region changes. TBD: iOS?
+    pub region: String,
+
+    /// The update channel of the user's build.
+    pub update_channel: SearchUpdateChannel,
+
+    /// The distribution id for the user's build.
+    pub distribution_id: String,
+
+    /// The search related experiment id that the user is included within. On
+    /// desktop this is the `searchConfiguration.experiment` variable.
+    pub experiment: String,
+
+    /// The application name that the user is using.
+    pub app_name: SearchApplicationName,
+
+    /// The application version that the user is using.
+    pub version: String,
+}
+
+/// Parameter definitions for search engine URLs. The name property is always
+/// specified, along with one of value, experiment_config or search_access_point.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchUrlParam {
+    /// The name of the parameter in the url.
+    pub name: String,
+    /// The parameter value, this may be a static value, or additionally contain
+    /// a parameter replacement, e.g. `{inputEncoding}`. For the partner code
+    /// parameter, this field should be `{partnerCode}`.
+    pub value: Option<String>,
+    /// The value for the parameter will be derived from the equivalent experiment
+    /// configuration value.
+    /// Only desktop uses this currently.
+    pub experiment_config: Option<String>,
+}
+
+/// Defines an individual search engine URL.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchEngineUrl {
+    /// The PrePath and FilePath of the URL. May include variables for engines
+    /// which have a variable FilePath, e.g. `{searchTerm}` for when a search
+    /// term is within the path of the url.
+    pub base: String,
+
+    /// The HTTP method to use to send the request (`GET` or `POST`).
+    /// If not specified, defaults to GET.
+    pub method: Option<String>,
```

Comment:
```suggestion
    pub method: String,
```

Idle thought: IIUC, this would leave it up to the JS / Kotlin / Swift consumer to null-check, and fall back to `GET` if the method is unspecified—is that right? I wonder if we can / should have the Rust logic pass back `GET` automatically?